### PR TITLE
Basic Functionality

### PR DESCRIPTION
**Pull request by max-moser**
# Conversion Between maDMPs and Records

Implemented logic for parsing maDMPs according to the [RDA Common Standard](https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard), and using the parsed information to create record drafts in Invenio.

Custom subclasses of the **BaseRecordConverter** can be used to implement conversion into/from custom metadata models.
The default **RDMRecordConverter** can be used to convert between DMPs and records using the Invenio-RDM-Records metadata model.

# Model Classes

Added database-persistent objects for storing information regarding which records are linked to which DMPs.
Among other things, this helps in minimizing the amount of unnecessarily sent network requests by checking if the record in question is linked to any DMPs at all.

# REST API

Implemented simple REST API endpoints for allowing maDMP tools to communicate with Invenio.
Optional token-based auth. (via the "Authorization: Bearer" HTTP header, containing a shared secret) can be used to verify that the request was not forged.

# CLI Commands

Created simple CLI commands to import DMPs from maDMP JSON files.

# Tests

Added a few unit tests and some test data (example maDMPs from the [RDA GitHub repository](https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard), and randomly generated example records [using the current RDM-Records metadata model]).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -17,15 +17,46 @@
...
+
+
+Invenio module for integrating machine-actionable data management plans (maDMPs)
+according to the `Research Data Alliance Common Standard <https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard>`_.
+
+
+::
+
+  I can help
+        -- Android Minsky
```
Comment (mb-wali) [inline README.rst]:
```suggestion
```

File: invenio_madmp/config.py [hunk 2]
```diff
@@ -6,3 +6,49 @@
...
 """Invenio module for maDMP integration."""
+
+from typing import Dict, List
+
+from .convert.records.base import BaseRecordConverter
+from .convert.records.rdm_records import RDMRecordConverter
+from .licenses import KNOWN_LICENSES
+
+# for finding matching distributions
+MADMP_HOST_URL = "https://data.tuwien.ac.at"
```
Comment (mb-wali) [inline invenio_madmp/config.py:17]: In order to run the config variables [readthedocs](https://invenio-madmp.readthedocs.io/en/latest/configuration.html), the comment should be something like this suggestion.
```suggestion
# for finding matching distributions
MADMP_HOST_URL = "https://data.tuwien.ac.at"
"""Host URL for MADMPs."""
```

**Review by mb-wali — COMMENTED** [review]
That's a very good start for a basic integration.
I left some comments, but overall nice work. I liked the code style black ```badge```. we will use it on other modules.
I haven't gone to the func in detail. since at this stage the module is using most of hard-coded values.
also, I noticed some default cookie-cutter setup that this module would not need.
* remove files/entrypoints for translations
* remove todos for example [here](https://github.com/fair-data-austria/invenio-madmp/blob/b1da22a3145ee29e731c97e2da90fbf86e9a583d/MANIFEST.in#L8)

maybe we can open a separate issue for it.

File: README.rst (hunk 1 above)
Comment (max-moser) [reply README.rst]: I removed the quote along with the reST start-of-quote marker.

File: invenio_madmp/config.py (hunk 2 above)
Comment (max-moser) [reply invenio_madmp/config.py:17]: created issue #9 for this

**Pull request merged**